traffic_ops/traffic_ops_golang/invalidationjobs/invalidationjobs.go

```diff
@@ -282,6 +493,152 @@ func (job *InvalidationJob) Read(h http.Header, useIMS bool) ([]interface{}, err
 
 // Used by POST requests to `/jobs`, creates a new content invalidation job
 // from the provided request body.
+func CreateV40(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	job := tc.InvalidationJobCreateV40{}
+	if err := json.NewDecoder(r.Body).Decode(&job); err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("Unable to parse Invalidation Job"), fmt.Errorf("parsing jobs/ POST: %v", err))
+		return
+	}
+
+	// Check if request object is valid
+	w.Header().Set(rfc.ContentType, rfc.ApplicationJSON)
+	if err := job.Validate(inf.Tx.Tx); err != nil {
+		response := tc.Alerts{
+			Alerts: []tc.Alert{
+				{
+					Text:  err.Error(),
+					Level: tc.ErrorLevel.String(),
+				},
+			},
+		}
+
+		resp, err := json.Marshal(response)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("Encoding bad request response: %v", err))
+			return
+		}
+
+		w.WriteHeader(http.StatusBadRequest)
+		api.WriteAndLogErr(w, r, append(resp, '\n'))
```

Comment:
This should probably just be a one-liner that does the appending for you:
```go
api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, nil, nil)
```
Honestly I don't think `api.WriteAndLogErr` should even be exported